packages/core/src/controllers/TransactionsController.ts

```diff
@@ -0,0 +1,93 @@
+import { DateUtil, type Transaction } from '@web3modal/common'
+import { proxy, subscribe as sub } from 'valtio/vanilla'
+import { BlockchainApiController } from './BlockchainApiController.js'
+import { OptionsController } from './OptionsController.js'
+import { AccountController } from './AccountController.js'
+
+// -- Types --------------------------------------------- //
+type TransactionByYearMap = Record<number, Transaction[]>
+
+export interface TransactionsControllerState {
+  transactions: Transaction[]
+  transactionsByYear: TransactionByYearMap
+  loading: boolean
+  empty: boolean
+  next: string | null
+}
+
+// -- State --------------------------------------------- //
+const state = proxy<TransactionsControllerState>({
+  transactions: [],
+  transactionsByYear: {},
+  loading: false,
+  empty: false,
+  next: null
+})
+
+// -- Controller ---------------------------------------- //
+export const TransactionsController = {
+  state,
+
+  subscribe(callback: (newState: TransactionsControllerState) => void) {
+    return sub(state, () => callback(state))
+  },
+
+  async fetchTransactions() {
+    const projectId = OptionsController.state.projectId
+    const accountAddress = AccountController.state.address
+
+    if (!projectId || !accountAddress) {
+      throw new Error("Transactions can't be fetched without a projectId and an accountAddress")
+    }
+
+    state.loading = true
+    await BlockchainApiController.fetchTransactions({
+      account: accountAddress,
+      projectId,
+      cursor: state.next
+    })
+      .then(response => {
```

Comment:
Split out into 2 awaits and wrap in `try / catch / finaly` block to follow other async actions style.

Reply:
@0xAsimetriq Can you check this again please?